services/src/main/java/org/exoplatform/addons/gamification/storage/AnnouncementStorage.java

```diff
@@ -47,6 +48,19 @@ public Announcement saveAnnouncement(Announcement announcement) {
     return EntityMapper.fromEntity(announcementEntity);
   }
 
+  public Announcement cancelAnnouncement(Announcement announcement) {
+    if (announcement == null) {
+      throw new IllegalArgumentException("Announcement argument is null");
+    }
+    RuleEntity ruleEntity = ruleDAO.find(announcement.getChallengeId());
+    GamificationActionsHistory announcementEntity = EntityMapper.toEntity(announcement, ruleEntity);
+    announcementEntity.setStatus(HistoryStatus.CANCELED);
+    announcementEntity.setActivityId(null);
+    announcementEntity.setObjectId(null);
+    announcementEntity = announcementDAO.update(announcementEntity);
+    return EntityMapper.fromEntity(announcementEntity);
+  }
+
```

Comment:
Better reuse what the `reject` action do instead of creating another parallel algorithm that do the same. (Reusing behaviors instead of recreating them)

```suggestion
```